fanficfare/adapters/adapter_silmarillionwritersguildorg.py

```diff
@@ -0,0 +1,236 @@
+        BaseSiteAdapter.__init__(self, config, url)
+
+        self.username = "NoneGiven" # if left empty, site doesn't return any message at all.
+        self.password = ""
+        self.is_adult=False
+
+        # get storyId from url--url validation guarantees query is only sid=1234
+        self.story.setMetadata('storyId',self.parsedUrl.query.split('=',)[1])
+
+        # normalized story URL.
+        self._setURL('http://' + self.getSiteDomain() + '/archive/home/viewstory.php?sid='+self.story.getMetadata('storyId'))
+
+        # Each adapter needs to have a unique site abbreviation.
+        self.story.setMetadata('siteabbrev','swg')
+
+        # The date format will vary from site to site.
+        # http://docs.python.org/library/datetime.html#strftime-strptime-behavior
+        self.dateformat = "%B %d, %Y"
+
+    @staticmethod # must be @staticmethod, don't remove it.
+    def getSiteDomain():
+        # The site domain.  Does have www here, if it uses it.
+        return 'www.silmarillionwritersguild.org'
+
+    @classmethod
+    def getSiteExampleURLs(cls):
+        return "https://"+cls.getSiteDomain()+"/archive/home/viewstory.php?sid=123"
+
+    def getSiteURLPattern(self):
+        return r"https?://"+re.escape(self.getSiteDomain()+"/archive/home/viewstory.php?sid=")+r"\d+$"
+
+    ## Getting the chapter list and the meta data, plus 'is adult' checking.
+    def extractChapterUrlsAndMetadata(self):
+
+        # index=1 makes sure we see the story chapter index.  Some
+        # sites skip that for one-chapter stories.
+        url = self.url
+        logger.debug("URL: "+url)
+
+        try:
+            data = self._fetchUrl(url)
+        except HTTPError as e:
+            if e.code == 404:
+                raise exceptions.StoryDoesNotExist(self.url)
+            else:
+                raise e
+
+        # use BeautifulSoup HTML parser to make everything easier to find.
+        soup = self.make_soup(data)
+        # print data
+
+        # Now go hunting for all the meta data and the chapter list.
+
+        ## Title and author
+        a = soup.find('h6')
+
+        titlelinks = a.find_all('a')
+        aut= titlelinks[1]
+        
+        self.story.setMetadata('authorId',aut['href'].split('=')[1])
+        self.story.setMetadata('authorUrl','https://'+self.host+'/archive/home/'+aut['href'])
+        self.story.setMetadata('author',aut.string)
+        asoup = self.make_soup(self._fetchUrl(self.story.getMetadata('authorUrl')))
+
+        self.story.setMetadata('title',a.find('strong').find('a').get_text())
+        
+        try:
+            # Find Series name from series URL.
+            a = soup.find('a', href=re.compile(r"viewseries.php\?seriesid=\d+"))
+            series_name = a.string
+            series_url = 'https://'+self.host+'/archive/home/'+a['href']
+
+            # use BeautifulSoup HTML parser to make everything easier to find.
+            seriessoup = self_make_soup(self._fetchUrl(series_url))
```

Comment:
I've fixed the series parsing but the site does some weird pagination on the seriesview.php page when there are more than 25 items in a series, causing issues with the series index not being filled.
To avoid this I've not attempted parsing for the series index

Reply:
Can you provide a series URL that shows that?  I'd like to see what it looks like.

Reply:
Nevermind, there's a top ten list for largest series.

I won't require it, but if you're interested, [here is an example](https://github.com/JimmXinu/FanFicFare/blob/master/fanficfare/adapters/adapter_efpfanficnet.py#L176) of a similar site doing similar pagination for authors that could be adapted for series.  It would also need to count story links to get the number, of course.

FFF has other adapters that can have to fetch multiple pages sometimes, but I'm not finding one for and eFiction site off hand.

It looks like silmarillionwritersguild.org similarly paginates author pages, but because you're not using anything from the author pages, it doesn't matter.  

It does reveal one last issue tho:
`asoup = self.make_soup(self._fetchUrl(self.story.getMetadata('authorUrl')))` isn't used and should be removed: [Here](https://github.com/JimmXinu/FanFicFare/pull/492/files#diff-56000d67aa713a2379182e499882e1acR101)

Reply:
Okay, I've implemented series index parsing and removed the redundant code

Reply:
Okay, thanks.  There's a performance tweak that can be done to series parsing(stop fetching pages after story is found), but I'll add that rather than go back and forth again.